packages/data-context/src/actions/WizardActions.ts

```diff
@@ -71,17 +71,19 @@ export class WizardActions {
 
     assert(currentTestingType)
     assert(chosenLanguage)
-    assert(!this.ctx.lifecycleManager.isTestingTypeConfigured(currentTestingType), `Cannot call this if the testing type (${currentTestingType}) has not been configured`)
+
     switch (currentTestingType) {
       case 'e2e': {
         this.ctx.coreData.scaffoldedFiles = await this.scaffoldE2E()
         this.ctx.lifecycleManager.refreshMetaState()
+        this.ctx.coreData.forceReconfigureProject = false
```

Comment:
Minor: I'm seeing the `ctx.update` function used more and more. Is this something we should make standard and apply to all updates of coreData, including this update?

Reply:
Yes we should! It took me a while to grok how this all is supposed to work, I started documenting it here. https://github.com/cypress-io/cypress/tree/10.0-release/packages/data-context#actions Please help if you see or know some better guidelines. Basically if everything can mutate via a single place, like `update`, we will theoretically be able to "play back" a session - good for debugging.

There are still some unknowns, but I'd like to get the workflow more formal.

Basically:

- coreData: the data
- sources: read/derive coreData, readonly
- actions: mutate data

Kind of like in Redux...

- state: immutable
- reducer: update state
- selector: derive state